### ISSUE TITLE
Roslyn doesn't recognize named constants when checking parameters' states on return

### ISSUE THREAD

**Issue by TessenR** [dotnet/roslyn#43222]
**Version Used**: 

```
commit ff3d2a75dcd9ad286d4bad47c993454f3d1dd1d7
Merge: 7be257abe9 6fd1f755aa
Author: msftbot[bot] <48340428+msftbot[bot]@users.noreply.github.com>
Date:   Thu Apr 2 15:49:03 2020 +0000

    Merge pull request #43000 from mavasani/PortConvertAnonymousTypeToTuple

    Port ConvertAnonymousTypeToTuple to shared layer
```

**Steps to Reproduce**:

```cs
#nullable enable
using System.Diagnostics.CodeAnalysis;

class C
{
  const bool True = true;
  public bool M1([NotNullWhen(false)] ref string? s)
  {
    return True; // warning
  }
    
  public bool M2([NotNullWhen(false)] ref string? s)
  {
    if (True)
        return true; // no warnings
  }
}
```
**Expected Behavior**:
No warnings. Both methods never return `false`

**Actual Behavior**:
`warning CS8762: Parameter 's' must have a non-null value when exiting with 'false'.` is reported in `M1`

**Comment by jcouv** [comment]
Looks like there is no longer a warning ([sharplab](https://sharplab.io/#v2:CYLg1APgxAdgrgGwQQwEYIKYAIMzZgWACgABABixIEYA6AEQEtkBzGAewGcAXBgYw5oBhNsAwBBPAgCeHBhwDcxYiQBMWQcQDexLFl5sY3LKjZsEWACoAnONgC8WLjYyKiukgGZjp8wFkqABQA2gBybFwhiAgA6gAWuAEAZsgIHBgAlAC6WFYYiZRUZAD8WBzpOljabrqUAOyWzvJYAPTNWADuyFYwDDDMFQC+FboVnt5mWL4qwWERUXEJyakZ2bn51MWl5dVVNVgM+QHWttt7eyT1TrZNrVjsHV09fRyDxANAA=)). 
I think this was fixed as part of change to enforcement of NotNullWhen.
